src/cnv/ugbio_cnv/combine_cnv_vcf_utils.py

```diff
@@ -264,43 +283,112 @@ def merge_cnvs_in_vcf(input_vcf: str, output_vcf: str, distance: int = 1000) ->
-            "CNMOPS_COHORT_STDEV",
-            "CopyNumber",
-        ]
-    }
-    all_fields = sum(action_dictionary.values(), [])
     update_df = vcftools.get_vcf_df(str(removed_vcf), custom_info_fields=all_fields + ["SVLEN", "MatchId"]).sort_index()
     update_df["matchid"] = update_df["matchid"].apply(lambda x: x[0]).astype(float)
     update_df["end"] = update_df["pos"] + update_df["svlen"].apply(lambda x: x[0]) - 1
+
+    if do_not_merge_collapsed_filtered:
+        unselect = (update_df["filter"] != "PASS") & (update_df["filter"] != "") & (update_df["filter"] != ".")
+        update_df = update_df.loc[~unselect]
+
     with pysam.VariantFile(output_vcf_collapse) as vcf_in:
         hdr = vcf_in.header
         with pysam.VariantFile(output_vcf, "w", header=hdr) as vcf_out:
             for record in vcf_in:
                 if "CollapseId" in record.info:
                     cid = float(record.info["CollapseId"])
                     update_records = update_df[update_df["matchid"] == cid]
-                    record.stop = update_records["end"].max()
-                    for action, fields in action_dictionary.items():
-                        for field in fields:
-                            values = np.array(list(update_records[field.lower()]) + [record.info[field]])
-                            if action == "weighted_avg":
-                                lengths = np.array(
-                                    list(update_records["svlen"].apply(lambda x: x[0])) + [record.info["SVLEN"][0]]
-                                )
-                                weighted_avg = np.sum(values * lengths) / np.sum(lengths)
-                                record.info[field] = round(weighted_avg, 3)
+                    if not update_records.empty:
+                        record.start = min(update_records["pos"].min(), record.start)
+                        record.stop = max(update_records["end"].max(), record.stop)
+
+                        for action, fields in AGGREGATION_ACTIONS.items():
+                            for field in fields:
+                                if field in hdr.info:
+                                    _value_aggregator(
+                                        record,
+                                        update_records,
+                                        field,
+                                        action,
+                                        hdr.info[field].number,
+                                        hdr.info[field].type,
+                                    )
 
                     record.info["SVLEN"] = (record.stop - record.start,)
+                collapse_info_fields = ["CollapseId", "NumCollapsed", "NumConsolidated"]
+                for c in collapse_info_fields:
+                    if c in record.info:
+                        del record.info[c]
                 vcf_out.write(record)
     mu.cleanup_temp_files(temporary_files)
+
+
+AGGREGATION_ACTIONS = {
```

Comment:
maybe this block should be in the beginning of the script or private inside merge_cnvs_in_vcf function.